git-utimes: Change files modification time to their last commit date

vt-alt opened the pull request:
New tool to change files modification time to their last commit date.

SuperSandro2000 commented on Commands.md:
````diff
@@ -1495,3 +1496,11 @@ Opens the current git repository website in your default web browser.
+Change files modification time to their last commit date.
+
+```bash
+## git utimes
````
```suggestion
$ git utimes
```

SuperSandro2000 commented on bin/git-utimes:
```diff
@@ -0,0 +1,25 @@
+else
+  # `-n` should be limited or parallelization will not give effect,
+  # because all args will go into single worker.
+  NPROC=$(nproc)
```
That does not work on FreeBSD. See https://unix.stackexchange.com/questions/108821/number-of-cores-on-freebsd

vt-alt commented:
- Fixed `Commands.md`.
- Portable nproc oneliner is from https://stackoverflow.com/a/53427092/1478629
- Added additional safeguard in case `NPROC` is still null.

vt-alt commented:
Ping! I fixed issues @SuperSandro2000 mentioned. Whats next?

spacewander commented on Commands.md:
````diff
@@ -1495,3 +1496,11 @@ Opens the current git repository website in your default web browser.
+Change files modification time to their last commit date.
+
+```bash
+$ git utimes
````
Need to show the difference before / after running this command.

spacewander commented on bin/git-utimes:
```diff
@@ -0,0 +1,25 @@
+if [ "$1" = --touch ]; then
+  # Internal use option only just to parallelize things.
+  shift
+  for f; do
```
Is the for loop required?

vt-alt replied:
`for` is to support multiple arguments. Why not? This should reduce number of `exec`s. Are you against of it?

vt-alt replied:
Correction: s/execs/forks/

spacewander replied:
I got it.

spacewander commented on bin/git-utimes:
```diff
@@ -0,0 +1,25 @@
+      touch -d "$t" "$f"
+    fi
+  done
+  exit
```
The `exit` is redundant.

spacewander commented on bin/git-utimes:
```diff
@@ -0,0 +1,25 @@
+# Change files modification time to their last commit date
+#
+# Copyright (c) 2020 Vitaly Chikunov <vt@altlinux.org>.
+# SPDX-License-Identifier: MIT
```
Please don't add the Copyright and License. We already have and only have a LICENSE file.

spacewander commented on bin/git-utimes:
```diff
@@ -0,0 +1,25 @@
+if [ "$1" = --touch ]; then
+  # Internal use option only just to parallelize things.
+  shift
+  for f; do
```
@vt-alt
I got it, it is OK.

vt-alt commented:
Force-pushed changes you requested - extended example in `Commands.md`, removed guard `exit`, removed license plate.

vt-alt commented:
Almost three weeks passed since all requested changes are applied. Is there anything to add?

spacewander commented:
@vt-alt 
Sorry for my delay!

vt-alt commented:
@spacewander Thanks!
